packages/components/src/range-control/index.tsx

```diff
@@ -121,16 +129,16 @@ function RangeControl(
 
 	const id = useInstanceId( RangeControl, 'inspector-range-control' );
 	const describedBy = !! help ? `${ id }__help` : undefined;
-	const enableTooltip = showTooltipProp !== false && isFinite( value );
+	const enableTooltip = hasTooltip !== false && isFinite( value );
 
-	const handleOnRangeChange = ( event ) => {
+	const handleOnRangeChange = ( event: ChangeEvent< HTMLInputElement > ) => {
 		const nextValue = parseFloat( event.target.value );
 		setValue( nextValue );
 		onChange( nextValue );
 	};
 
-	const handleOnChange = ( nextValue ) => {
-		nextValue = parseFloat( nextValue );
+	const handleOnChange = ( nextValue: number | string | null ) => {
+		nextValue = parseFloat( nextValue as string );
```

Comment:
We should only `parseFloat` when `nextValue` is a `string`

```suggestion
		nextValue =
			typeof nextValue === 'string' ? parseFloat( nextValue ) : nextValue;
```

Reply:
I've tried to tidy this up and make the `handleOnChange` and `handOnReset` more consistent. Let me know if this is sufficient or what else you had in mind. (https://github.com/WordPress/gutenberg/pull/40535/commits/b17df9a2bacc8075b6df7097678edee91514b371)